### PR TITLE
Add enhanced alerting details to host monitoring docs

### LINKED ISSUE #3747: [Request]: Enhanced Alerting in Hosts
### Description

In 8.14 we'll be providing better alerting capabilities in [Hosts](https://edge-lite-oblt.kb.us-west2.gcp.elastic-cloud.com/app/metrics/hosts?_a=(dateRange:(from:now-15m,to:now),filters:!(),limit:500,panelFilters:!(),query:(language:kuery,query:%27%27))&controlPanels=(cloud.provider:(explicitInput:(fieldName:cloud.provider,id:cloud.provider,title:%27Cloud%20Provider%27),grow:!f,order:1,type:optionsListControl,width:medium),host.os.name:(explicitInput:(fieldName:host.os.name,id:host.os.name,title:%27Operating%20System%27),grow:!f,order:0,type:optionsListControl,width:medium),service.name:(explicitInput:(fieldName:service.name,id:service.name,title:%27Service%20Name%27),grow:!f,order:2,type:optionsListControl,width:medium))).

Namely:

1. Active alert counts per host

**1. Active alert counts per host**

This is a new column:

image

**2. More alert detail per host**

Previously, we just showed 'active alert count' - no table or chart etc:

_With active alerts_
image

_Without active alerts_
image

> **Note** : We have a new 'collapsible' functionality which users can use (and we use to collapse unnecessary sections)

### Resources

[Parent issue](https://github.com/elastic/observability-dev/issues/3304)

### Which documentation set does this change impact?

Stateful and Serverless

### Feature differences

None

### What release is this request related to?

8.14

### Collaboration model

The documentation team

### Point of contact.

**Main contact:** @roshan-elastic @crespocarlos 

**Stakeholders:**

### PR DESCRIPTION
**Pull request by dedemorton**
Closes #3747 and #3627.

- [x] Newbie question: It's possible I don't understand this feature fully, but it seems like users need to group by hostname when they create the alert, or the alert will not appear in the Hosts view. Is this correct? If so, I think we should mention that somewhere. Since this setting is optional, users might not realize it's required if they want to view alerts for specific hosts. WDYT? Am I missing something here?

  image

TODO (after merging):

- [ ] Port to serverless docs.

### CODE REVIEW

**Comment by dedemorton** [comment]
@roshan-elastic @crespocarlos Can you take a look when you have a chance and respond to my question here? Thanks! (Sorry didn't mean to add your names to the related issue. Just got my browser tabs messed up.)

**Comment by roshan-elastic** [comment]
Hey @dedemorton 

> Closes #3747 and #3627.
> 
> Newbie question: It's possible I don't understand this feature fully, but it seems like users need to group by hostname when they create the alert, or the alert will not appear in the Hosts view. Is this correct? If so, I think we should mention that somewhere. Since this setting is optional, users might not realize it's required if they want to view alerts for specific hosts. WDYT? Am I missing something here?
> 
> image
> 
> TODO (after merging):
> 
> * [ ]  Port to serverless docs.

Yeah, that's right. There are pretty much two ways for alerts to show up against hosts:

(1) They use the 'Inventory' rule and select the 'Host'

image

(2) If they use a rule which is possible to 'group by' `host.name` then it should show as well

e.g. metric threshold rules, custom threshold rules [beta]

_Sample Metrics Threshold Rule_
[screenshot](https://github.com/elastic/observability-docs/assets/117740680/1cbb7654-8e2c-47a9-9ba0-22d6a20e04e3)

_Sample Custom Threshold Rule_
[screenshot](https://github.com/elastic/observability-docs/assets/117740680/c9836247-279b-47a0-a3bd-c490f96985b8)

However, let me double-check with the engineers on this.

@crespocarlos (cc @jennypavlova) - When I test the custom threshold and metric threshold rules:

1. Metric Threshold doesn't appear to fire at all (so not sure if this will show as an alert in the host view)
2. The custom threshold rule fires but doesn't show in the host view...is this because we only show 'inventory' and 'metric threshold' rules?

[Example Host](https://edge-lite-oblt.kb.us-west2.gcp.elastic-cloud.com/app/metrics/hosts?_a=(dateRange:(from:now-15m,to:now),filters:!(),limit:50,panelFilters:!(),query:(language:kuery,query:%27%27))&controlPanels=(cloud.provider:(explicitInput:(fieldName:cloud.provider,id:cloud.provider,title:%27Cloud%20Provider%27),grow:!f,order:1,type:optionsListControl,width:medium),host.os.name:(explicitInput:(fieldName:host.os.name,id:host.os.name,title:%27Operating%20System%27),grow:!f,order:0,type:optionsListControl,width:medium),service.name:(explicitInput:(fieldName:service.name,id:service.name,title:%27Service%20Name%27),grow:!f,order:2,type:optionsListControl,width:medium))&tableProperties=(detailsItemId:gke-edge-lite-oblt-edge-lite-oblt-poo-3a29949c-ljqp-Ubuntu,pagination:(pageIndex:0,pageSize:10),sorting:(direction:desc,field:alertsCount))&assetDetails=(dateRange:(from:now-15m,to:now),name:gke-edge-lite-oblt-edge-lite-oblt-poo-3a29949c-ljqp))

_Only one alert showing in the host view_
image

Here are the alerts set up:

See [alert rules](https://edge-lite-oblt.kb.us-west2.gcp.elastic-cloud.com/app/observability/alerts/rules?_a=(lastResponse:!(),params:(),search:%27%27,status:!(),type:!()))
<br>
image

**Question**
- Are you able to confirm the logic on which alerts show against a host (and how we determine that)?

**Comment by crespocarlos** [comment]
Hey @roshan-elastic 

> - Metric Threshold doesn't appear to fire at all (so not sure if this will show as an alert in the host view)

It was using `host.cpu.pct` field, which has no value. Was this intentional?

image

After adjusting it to use `system.cpu.system.norm.pct` field, the alert fired and appeared in the hosts view.
image


> - The custom threshold rule fires but doesn't show in the host view...is this because we only show 'inventory' and 'metric threshold' rules?

Yeah. We might need to adjust something in the custom threshold executor for it to show up in the alerts table. The actionable observability owns that, but I guess we can submit a PR.

**Comment by dedemorton** [comment]
>We might need to adjust something in the custom threshold executor for it to show up in the alerts table.

@crespocarlos Is there something I can say in the docs to help users? Maybe something like, "To see alerts on the Hosts page, the rule triggering the alert must either select hosts in a condition or be configured to group by hostname." Is this statement correct?

**Comment by crespocarlos** [comment]
Hi @dedemorton , regarding Custom Threshold it's something we need to fix in the code first, but it will work the same way as  `Metrics Threshold`.

The conditions for users to see alerts in the Hosts view are - Similar to what you wrote, but it might help to specify what needs to be done depending on the rule type?

Metrics Threshold - When creating the alert, inform` host.name` on  the `group by` field 
Custom Threshold - When creating the alert, inform` host.name` on  the `group by` field  (When fixed)
Inventory - Select `Host` in `Condition`

**Comment by dedemorton** [comment]
@roshan-elastic @crespocarlos I've updated the documentation. Can you confirm my changes and approve the topic if you think it's ready to merge? Thanks!

File: docs/en/observability/monitor-infra/analyze-hosts.asciidoc
Comment (dedemorton) [inline docs/en/observability/monitor-infra/analyze-hosts.asciidoc:179]:
```suggestion
edit the rules and make sure they are correctly configured to associate the host name with the alert:
```

**Comment by roshan-elastic** [comment]
> It was using host.cpu.pct field, which has no value. Was this intentional?

@crespocarlos - 🤦 thanks! Even I can't use ECS haha

**Comment by dedemorton** [comment]
@roshan-elastic Is this PR ready to merge? I still need SME approval. Thanks!

**Review by roshan-elastic — APPROVED** [review]
Sorry @dedemorton - completely missed this.

Might be too late for this version but approving for whenever it can go in

**Review by bmorelli25 — APPROVED** [review]

**Pull request merged**
